### PR TITLE
Fix `data_write()` for partially labelled characters

### PR DESCRIPTION
**Pull request by strengejacke**
@etiennebacher We should merge this fix before submiting _datawizard_.

### CODE REVIEW

File: tests/testthat/test-data_write.R [hunk 1]
```diff
@@ -132,3 +132,19 @@ test_that("data_write, no file extension", {
...
+
+test_that("data_write, no file extension", {
+  d <- data.frame(
+    a = letters[1:3],
+    stringsAsFactors = FALSE
+  )
+  d$a <- assign_labels(d$a, variable = "First")
+  # expect message, but no error
+  expect_message(data_write(d, "test.sav"), regex = "Preparing")
+})
```
Comment (etiennebacher) [inline tests/testthat/test-data_write.R:150]: do you want to check by reading the data back as in the other tests above?

File: tests/testthat/test-data_write.R [hunk 2]
```diff
@@ -132,3 +132,19 @@ test_that("data_write, no file extension", {
...
   expect_error(data_write(d, NULL))
 })
+
+
+# writing character vector works for missing value labels ------------------
+
+tmp <- tempfile(fileext = ".sav")
+on.exit(unlink(tmp))
+
+test_that("data_write, no file extension", {
```
Comment (etiennebacher) [inline tests/testthat/test-data_write.R:142]:
```suggestion
test_that("data_write, existing variable label but missing value labels", {
```

**Review by etiennebacher — APPROVED** [review]

File: tests/testthat/test-data_write.R (hunk 1 above)
Comment (etiennebacher) [reply tests/testthat/test-data_write.R:150]: Ok, I'm fine with this then

**Comment by strengejacke** [comment]
Working with real-life data is the best stress-test ;-)

**Pull request merged**